database/migrations/2020_03_22_1424_add_pkey_to_device_graphs.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+    This migration adds primary key for device_graphs
+
+    Percona Xtradb refused to INSERT IGNORE into a table
+    without a primary key.
+ */
+
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+
+class AddPrimaryKeyToDeviceGraphs extends Migration
+{
+
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        DB::table('device_graphs')->delete();
```

Comment:
This can't be right?

Reply:
Actually, that table might contain duplicates, as no constraint prevented that. Thus, clear the table, then add the primary key. Also, poller modules will fill the table upon next run. Am I right?

Reply:
@rkojedzinszky you can run through and delete duplicates first.

Reply:
Unfortunately, on a table without a primary key, you cannot delete just duplicates. You cannot identify them :)

Reply:
Add a new column with a primary key first then.

Reply:
Unfortunately, I am not a php developer, and perhaps, a bigger refactor would be needed here, which I cannot handle. The original problem involves a table without primary key and inserting into it with `insert ignore`. This breaks librenms on percona xtradb. Perhaps, the real solution is somewhere else.